### ISSUE TITLE
Odd performance difference

### ISSUE THREAD

**Issue by zenna** [zenna/Omega.jl#76]
Given
```julia
x = uniform(-1, 1)
y = uniform(-100, 100)
p1 = x ==ₛ 0
p2 = y ==ₛ 0
```

Then this:
```julia
julia> withkernel(kseα(100000)) do 
         samples = rand((p1, p2, x, y), p1 & p2, 10000, alg = NUTS)
       end
```

is generally much slower than this:

```julia
julia> withkernel(kseα(100000)) do 
         samples = rand((p1, p2), p1 & p2, 10000, alg = NUTS)
       end
```

And there's no reason why that should be the case.

Hunch

**Comment by jburroni** [comment]
was this closed by #86 ? 
The first snippet takes: ` 4.652043s` while the second takes `4.062395s`.
NUTS is currently using ForwardMode AD, which means that it needs `n` sweeps to compute the gradients, where `n` is the number of variables. For the second code, you have `n = 2 +2` (counting momentum), whereas for the first one you have `n=4+4` and so the difference can be explained by that.

**Comment by zenna** [comment]
Yeah, before it was a lot slower.  I think it's memoization that fixed it, but perhaps the reason it was slow is what you suggested.
